projects/services-ogc/src/lib/owc/owc-json.service.ts

```diff
@@ -331,195 +460,256 @@ export class OwcJsonService {
     return (offering.contents && !offering.operations) ? true : false;
   }
 
-  getOfferingContents(offering: IOwsOffering): IOwsOperation[] | IOwsContent[] {
-    if (this.checkIfServiceOffering(offering)) {
-      return offering.operations;
-    } else if (this.checkIfDataOffering(offering)) {
-      return offering.contents;
-    }
-  }
-
   /**
    * Helper function to extract legendURL from project specific ows Context
    * @param offering layer offering
    */
   getLegendUrl(offering: IOwsOffering) {
     let legendUrl = '';
 
-    if (offering.hasOwnProperty('styles')) {
+    if (offering.styles) {
       const defaultStyle = offering.styles.find(style => style.default);
       if (defaultStyle) {
         return defaultStyle.legendURL;
       }
-    } else if (offering.hasOwnProperty('legendUrl')) {
+    } else if (offering.legendUrl) {
       legendUrl = offering.legendUrl;
     }
     return legendUrl;
   }
 
-  /**
-   * retrieve iconUrl based on IOwsOffering
-   */
-  getIconUrl(offering: IOwsOffering) {
-    let iconUrl = '';
-    if (offering.hasOwnProperty('iconUrl')) {
-      iconUrl = offering.iconUrl;
-    }
-    return iconUrl;
-  }
 
   /**
-   * layer priority: first wms, then wmts, then wfs, then others.
+   * Get all Layers from the IOwsContext.
+   *
+   * The order of the layers is reversed to get the context drawing order!
    */
   getLayers(owc: IOwsContext, targetProjection: string): Observable<(Layer | LayerGroup)[]> {
-    const resources = owc.features;
     const layers$: Observable<Layer | LayerGroup>[] = [];
+    /** For the order of Layers see IOwsContext['features'] */
+
+    /**
+     * LayerGroups
+     *
+     * e.g. if groupName: Layers/test -> a group "test" in the slot Layers will be created with the layer in it
+     * e.g. if groupName: Overlays/test -> a group "test" in the slot Overlays will be created with the layer in it
+     * if groupName is only: Layers | Overlays | Baselayers use layerResources
+     */
+
+    const resources = this.getResources(owc);
+    const groups = [];
+    resources.forEach(r => {
+      const lg = this.createLayerOrGroupFromResource(r, owc, targetProjection, groups);
+      layers$.push(lg);
+    });
 
-    resources.sort((r1, r2) => r1.properties.folder > r2.properties.folder ? 1 : -1);
-
-    for (let i = 0; i < resources.length; i++) {
-      const resource = resources[i];
+    return forkJoin(layers$)
+      // making sure no undefined/null layers are returned
+      .pipe(map(layers => layers.filter(layer => layer)))
+      // reverse so layer order is like in the context
+      .pipe(map(layers => layers.reverse()));
```

Comment:
Here we did a reverse because the order of OWC Context is the other way around like our normal ukis layers work.
See 
- addUpdateLayers: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/service-owc/projects/map-ol/src/lib/map-ol.component.ts#L156 
and 
- setUkisLayers: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/58bb7549de9cdc7b38022d6a000d5401797bc183/projects/map-ol/src/lib/map-ol.service.ts#L662